media-tv/mythtv/mythtv-30.0_p20190808.ebuild

```diff
@@ -0,0 +1,406 @@
+
+Mythtv is updated to use correct FHS/Gentoo policy paths.
+Updating mythtv installations may report:
+	* mythtv is in use, cannot update home
+	* There was an error when attempting to update the home directory for mythtv
+	* Please update it manually on your system (as root):
+	*       usermod -d "/var/lib/mythtv" "mythtv"
+This can be ignored. The previous default was "/home/mythtv".
+Use caution if you change the home directory.
+
+To have this machine operate as recording host for MythTV,
+mythbackend must be running. Run the following:
+rc-update add mythbackend default
+
+Your recordings folder must be owned 'mythtv'. e.g.
+chown -R mythtv /var/lib/mythtv
+
+Want mythfrontend to start automatically?
+Set USE=autostart. Details can be found at:
+https://dev.gentoo.org/~cardoe/mythtv/autostart.html
+
+Note that the systemd unit now restarts by default and logs
+to journald via the console at the notice verbosity.
+"
+
+pkg_setup() {
+	python-single-r1_pkg_setup
+	# The acct-user/mythtv package creates/manages the user 'mythtv'
+}
+
+src_prepare() {
+	default
+
+	# Perl bits need to go into vender_perl and not site_perl
+	sed -e "s:pure_install:pure_install INSTALLDIRS=vendor:" \
+		-i "${S}"/bindings/perl/Makefile || die "Cannot convert site_perl to vendor_perl!"
+
+	# Fix up the version info since we are using the fixes/${PV} branch
+	echo "SOURCE_VERSION=\"v${MY_PV}\"" > "${S}"/VERSION
+	echo "BRANCH=\"${MYTHTV_BRANCH}\"" >> "${S}"/VERSION
+	echo "SOURCE_VERSION=\"${BACKPORTS}\"" > "${S}"/EXPORTED_VERSION
+	echo "BRANCH=\"${MYTHTV_BRANCH}\"" >> "${S}"/EXPORTED_VERSION
+
+	echo "setting.extra -= -ldconfig" >> "${S}"/programs/mythfrontend/mythfrontend.pro
+}
+
+src_configure() {
+	local -a myconf
+
+	# Setup paths
+	myconf+=(--prefix="${EPREFIX}"/usr)
+	myconf+=(--libdir="${EPREFIX}"/usr/$(get_libdir))
+	myconf+=(--libdir-name=$(get_libdir))
+	myconf+=(--mandir="${EPREFIX}"/usr/share/man)
+
+	if use debug; then
+		myconf+=(--compile-type=debug)
+		myconf+=(--disable-stripping) # FIXME: does not disable for all files, only for some
+		myconf+=(--enable-valgrind) # disables timeouts for valgrind memory debugging
+	else
+		myconf+=(--compile-type=release)
+	fi
+
+	# Build boosters
+	has ccache "${FEATURES}" || myconf+=(--disable-ccache)
+	has distcc "${FEATURES}" || myconf+=(--disable-distcc)
+
+	# CPU settings
+	### TODO set --arch and --tune too?
+	### This seems complicated is there something already done by archtecture masking?
+	### Bunch of other settings too. Ts this what flag-o-matic is used for?
+	for i in $(get-flag march) $(get-flag mcpu) $(get-flag mtune) ; do
+		[ "${i}" = "native" ] && i="host"
+		myconf+=(--cpu="${i}")
+		break
+	done
+	myconf+=($(use_enable altivec))
+
+	# Sound Output Support
+	myconf+=($(use_enable oss audio-oss))
+	myconf+=($(use_enable alsa audio-alsa))
+	myconf+=($(use_enable jack audio-jack))
+	myconf+=($(use_enable pulseaudio audio-pulseoutput))
+
+	# Input Support
+	myconf+=($(use_enable lirc))
```

Comment:
Just a note, you don't have to include `myconf+=` in every line here. You can just make it multi-line, like 
```
# Input Support
	myconf+=(
		$(use_enable lirc)
		$(use_enable input_devices_joystick joystick-menu)
		$(use_enable cec libcec)
		$(use_enable ieee1394 firewire)
	)
```
etc you get the idea, this applies to pretty much every section in this ebuild if you wish to make it cleaner.

Reply:
>     Can you walk me through what this does exactly? Besides march should include mtune so you don't need to check both.
I don't know why it is there. I'll look into it.

>  Just a note, you don't have to include myconf+= in every line here. You can just make it multi-line, like ...

I like your style much better.

Reply:
`Mythtv` uses the configure from `ffmpeg.` This bit of code is in the current `ffmpeg` ebuild. It tweaks some of the compiler optimizations.